arch/arm/src/stm32h7/Kconfig

```diff
@@ -284,11 +284,27 @@ config STM32H7_HAVE_SMPS
 	bool
 	default n
 
+choice
+	prompt "STM32 H7 Power Supply Selection"
+	default STM32H7_PWR_DEFAULT_SUPPLY
+	depends on ARCH_CHIP_STM32H7
+	---help---
+            Select the power supply mode of your board
+
+config STM32H7_PWR_DEFAULT_SUPPLY
+	bool "Default"
+
```

Comment:
Add help to all these  `config`s to describe what mode it is, based on the bits selected by `STM32 H7 Power Supply Selection`

Reply:
@davids5
We tried putting other modes into Kconfig.